arviz/stats/stats.py

```diff
@@ -1280,7 +1311,9 @@ def summary(
     n_vars = np.sum([joined[var].size // n_metrics for var in joined.data_vars])
 
     if fmt.lower() == "wide":
-        summary_df = pd.DataFrame(np.full((n_vars, n_metrics), np.nan), columns=metric_names)
+        summary_df = pd.DataFrame(
+            (np.full(cast(Tuple[int, int], (n_vars, n_metrics)), np.nan)), columns=metric_names
```

Comment:
why do we have to cast this to a tuple of ints? shouldn't it be one by construction? `n_metrics` is a length and `n_vars` is a sum over sizes integer divided by `n_metrics` :confused:

Reply:
Without that cast, I get this error:

```
arviz/stats/stats.py:1314: error: Argument 1 to "full" has incompatible type "Tuple[Union[number[Any], ndarray], int]"; expected "Union[int, Sequence[int]]"
```

mypy believes that `n_vars` could be an `ndarray`, because I believe it does not think deeply enough about what `np.sum()` could return.

I can move the `cast` so that it's just `cast(int, n_vars)`, but I can't get rid of it.  See my latest push.